lib/fsm/utils.go

```diff
@@ -139,6 +140,59 @@ func ResolvePlan(plan storage.OperationPlan, changelog storage.PlanChangelog) *s
 	return &plan
 }
 
+// DiffPlan returns the difference between the next and previous plans in the
+// form of a changelog.
+func DiffPlan(prevPlan, nextPlan storage.OperationPlan) (diff []storage.PlanChange, err error) {
+	// Quick sanity check that this is the same plan.
+	if prevPlan.OperationID != nextPlan.OperationID {
+		return nil, trace.BadParameter("can't diff different plans: %v %v", prevPlan, nextPlan)
+	}
+	// Since this is the same plan, should be safe to assume they have the
+	// same phases with different states.
+	prevPhases := prevPlan.GetLeafPhases()
```

Comment:
I'm not sure about this. At best this logic will just move elsewhere and will have to be handled there. But this method should still check for plans similarity, otherwise something might panic (e.g. if it blindly assumes the plans are the same) / not work as expected.